### PR TITLE
Fixing the issues in partition add with delete column as empty

### PR DESCRIPTION
**Pull request by zhljen**
Fixing the issues in partitionId attachment calculation when delete column is empty.

### CODE REVIEW

File: metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java [hunk 1]
```diff
@@ -140,9 +165,15 @@ public TablePartitionsUpdatePayload createTablePartitionsUpdatePayload(
      */
     public List<String> getSortedDeletionPartitionKeys(final List<PartitionDto> partitionDtos,
```
Comment (ajoymajumdar) [inline metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java]: Make this method access private if it is not being used elsewhere. Also remove the null check.

File: metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java [hunk 2]
```diff
@@ -140,9 +165,15 @@ public TablePartitionsUpdatePayload createTablePartitionsUpdatePayload(
...
+        //still check in case the function is used by other places
+        if (StringUtils.isEmpty(deleteColumn)) {
+            return Collections.emptyList();
+        }
+        final List<String> ret = partitionDtos.stream()
+            .map(x -> PartitionUtil.getPartitionKeyValues(x.getName().toString()).get(deleteColumn))
+            .filter(Objects::nonNull)
+            .sorted()
+            .collect(Collectors.toList());
         Collections.reverse(ret);
```
Comment (ajoymajumdar) [inline metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java]: You can also use sorted() method of streams.
..{
return partitionDtos.stream()
.map(x -> PartitionUtil.getPartitionKeyValues(x.getName().toString()).get(deleteColumn))
.filter(Objects::nonNull)
.sorted(Comparator.reversed())
.collect(Collectors.toList());
}

File: metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java [hunk 3]
```diff
@@ -91,37 +93,60 @@ public TablePartitionsUpdatePayload createTablePartitionsUpdatePayload(
...
+                    message = SNSNotificationPartitionAddMsg.NO_CANDIDATE_PARTITION_KEYS.name();
                 }
+            } else {
+                message = SNSNotificationPartitionAddMsg.MISSING_METADATA_INFO_FOR_PARTITION_KEY.name();
             }
-        } else {
-            message = SNSNotificationPartitionAddMsg.MISSING_METADATA_INFO_FOR_PARTITION_KEY.name();
+
+        } catch (Exception ex) {
+            message = SNSNotificationPartitionAddMsg.FAILURE_OF_GET_LATEST_PARTITION_KEY.name();
```
Comment (ajoymajumdar) [inline metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java]: Don't we want to publish atlas metrics about this?

File: metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java (hunk 1 above)
Comment (zhljen) [reply metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java]: The reason to make it public is to setup unit testing around it

File: metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java (hunk 2 above)
Comment (zhljen) [reply metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java]: Sounds good.

File: metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java (hunk 3 above)
Comment (zhljen) [reply metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java]: We have the atlas published.

File: metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java (hunk 1 above)
Comment (ajoymajumdar) [reply metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java]: Annotate the method with VisibleForTesting

Comment (zhljen) [reply metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java]: got it.

**Pull request merged**